packages/jest-diff/src/diffStrings.js

```diff
@@ -46,6 +48,20 @@ function diffStrings(a: string, b: string, options: ?DiffOptions): string {
       lines.pop();
     }
 
+    if (!part.added && !part.removed && linesLength > DIFF_TRIM * 2) {
+      return lines.map((line, idx) => {
+        if (idx === DIFF_TRIM) {
+          return chalk.yellow('@@ collapsed ')
+            + chalk.yellow(linesLength - DIFF_TRIM * 2)
+            + chalk.yellow(' unchanged lines\n');
```

Comment:
If they are all the same color, you only need to call chalk.yellow once.

Also return style is either:

```
return (
  chalk.yellow(…);
);
```

or 

```
return chalk.yellow(
  ...
);
```